python/GangaAtlas/Lib/Athena/Athena.py

```diff
@@ -1549,20 +1553,80 @@ def split(self,job):
 
             if (job.inputdata._name == 'ATLASCastorDataset') or \
                    (job.inputdata._name == 'ATLASLocalDataset'):
-                inputnames=[]
+                inputnames = []
+                outputnames = []
                 numfiles = len(job.inputdata.get_dataset_filenames())
                 if self.numfiles_subjob > 0:
-                    import math
                     self.numsubjobs = int( math.ceil( numfiles / float(self.numfiles_subjob) ) )
                 if self.match_subjobs_files:
-                    self.numsubjobs = numfiles 
-                for i in xrange(self.numsubjobs):    
-                    inputnames.append([])
+                    self.numsubjobs = numfiles
                 if numfiles < self.numsubjobs:
                     self.numsubjobs = numfiles
                     logger.warning('Number of requested subjobs larger than number of files found - changing j.splitter.numsubjobs = %d ' %self.numsubjobs )
-                for j in xrange(numfiles):
-                    inputnames[j % self.numsubjobs].append(job.inputdata.get_dataset_filenames()[j])
+
+                # check for output data mapping
+                if self.output_loc_to_input and isinstance(job.outputdata, ATLASOutputDataset):
+
+                    # check all input data is available in the mapping dictionary
+                    all_input = []
+                    for file_list in self.output_loc_to_input.values():
+                        all_input += file_list
+
+                    if frozenset(all_input) != frozenset(job.inputdata.names):
+                        raise ApplicationConfigurationError(None, 'Not all inputdata specified in splitter data mapping.'
```

Comment:
Unfortunately, I don't think so because the values in `output_loc_to_input` are lists, i.e.
```
[ [a, b], [c, d, e], ...]
```
and python tells me this isn't hashable 😄